### ISSUE TITLE
DebugPointer to work with generatePdf

### ISSUE THREAD

**Issue by blogcraft** [QuestPDF/QuestPDF#290]
I can't get DebugPointer to work. And in the docs it sayis it will be ignored with generatePdf.

I'm having DocumentLayoutException and I cant get the trace to work.

**Comment by Bebo-Maker** [comment]
Not sure whats your exact problem, but here is a small example which works fine for me.
```csharp
try
{
    Document
    .Create(container =>
    {
        container.Page(page =>
        {
            page.Size(PageSizes.A4.Landscape());

            page.Content()
                .Column(c =>
                {
                    c.Item().DebugPointer("Your debug pointer")
                               .Width(10000); //Forcing layout exception here to demonstrate debug pointer.
                });
        });
    })
    .GeneratePdf();
}
catch (DocumentLayoutException ex)
{
    Debug.WriteLine(ex.ElementTrace);
    throw;
}
```

`ex.ElementTrace` will container the trace information if a debugger is attached. Small excerpt of the ElementTrace with the debug pointer: 
```
🔥 DebuggingProxy
      -----------------
      Available space: (Width: 842,000, Height: 595,400)
      Requested space: Wrap

          🔥 Your debug pointer 🌟
          ------------------------
          Available space: (Width: 842,000, Height: 595,400)
          Requested space: Wrap

              🔥 Constrained
              --------------
              Available space: (Width: 842,000, Height: 595,400)
              Requested space: Wrap
              Min Width: 10000
              Max Width: 10000
              Min Height: -
              Max Height: -
```

**Comment by blogcraft** [comment]
Thanks! It did work that way!

I found out that a text was wraping but had no space.

Now I can see the trace. But it doesn't make sense...

Here is an exerpt (the trace was gigantic!):
```
🔥 DebuggingProxy
-----------------
Available space: (Width: 173,500, Height: 42,705)
Requested space: Wrap

    🔥 Row
    ------
    Available space: (Width: 173,500, Height: 42,705)
    Requested space: Wrap

        DebuggingProxy
        --------------
        Available space: (Width: 14.400,000, Height: 14.400,000)      ⬅️⬅️⬅️⁉️with and height came out of nowhere❗
        Requested space: FullRender (Width: 125,385, Height: 22,590)

            Column
            ------
            Available space: (Width: 14.400,000, Height: 14.400,000)
            Requested space: FullRender (Width: 125,385, Height: 22,590)

                DebuggingProxy
                --------------
                Available space: (Width: 14.400,000, Height: 14.400,000)
                Requested space: FullRender (Width: 125,385, Height: 11,295)

                    asesor 🌟
                    ---------
                    Available space: (Width: 14.400,000, Height: 14.400,000)
                    Requested space: FullRender (Width: 125,385, Height: 11,295)

                        DefaultTextStyle
                        ----------------
                        Available space: (Width: 14.400,000, Height: 14.400,000)
                        Requested space: FullRender (Width: 125,385, Height: 11,295)

                            Column
                            ------
                            Available space: (Width: 14.400,000, Height: 14.400,000)
                            Requested space: FullRender (Width: 125,385, Height: 11,295)

                                DebuggingProxy
                                --------------
                                Available space: (Width: 14.400,000, Height: 14.400,000)
                                Requested space: FullRender (Width: 125,385, Height: 11,295)

                                    TextBlock
                                    ---------
                                    Available space: (Width: 14.400,000, Height: 14.400,000)
                                    Requested space: FullRender (Width: 125,385, Height: 11,295)
                                    Alignment: Left
                                    Text: Tempore quibusdam dolorum
```

Also I had to cut down the trace because strangely many parts of it repeated and didn't make sense why.

**Comment by MarcinZiabek** [comment]
The 14400 points is a maximum page size in PDF. I expect that you have used the Row.AutoItem. The Row element needs to measure its AutoItem children to learn more about required size constraints, basically how big such items are and how much space should be reserved. The maximum space size is provided as an optimization (instead of inifinity).

**Comment by blogcraft** [comment]
@MarcinZiabek  thanks for the explanation. Yes, there is an AutoItem. 

```
column.Item().Row(row =>
{
    row.ConstantItem(50).Column(col =>
    {
        col.Item().Text("Asesor:");
        col.Item().Text("Número:");
    });
    row.AutoItem().Column(col =>
    {
        col.Item().DebugPointer("asesor").Text($"{Encabezado.Asesor}").Bold();
        col.Item().DebugPointer("numerocta").Text($"{Encabezado.NumeroCuenta}").Bold();
    });
});

```

Maybe there could be an improvement? The trace was very hard to follow and wasn't very clear where the problem was. Also, some items repeated many times in the trace even tho they appeared only once in the document (and also an infamous amount of `DebuggingProxy` items that its not clear what they represent).
